enum SendRecordType
{
    SELF_ACTIVE = 1;
    RESPONSE_REQUEST = 2;
}

enum NotificationType
{
    REQUEST_GIFT = 1;
    RECEIVE_GIFT = 2;
    NIL = 3; /* for lazy deletion */
}

message GoodsInfo
{
    required uint32 id = 1;
    required uint32 count = 2;
}

message SendRecord
{
    required uint32 uin = 1;
    required SendRecordType type = 2;
}

message ReceiveRecord
{
    required uint32 uin = 1;
}

message RequestRecord
{
    required uint32 uin = 1;
}

message Notification
{
    required uint32 uin = 1;
    required uint32 id = 2;
    required NotificationType type = 3;
    required uint32 create_time = 4;
    repeated GoodsInfo goods_list = 5;
}

message ExchangeGiftsInfo
{
    repeated SendRecord send_records = 1;
    repeated ReceiveRecord receive_records = 2;
    repeated RequestRecord request_records = 3;
    repeated Notification notifications = 4;
    required uint32 uin = 5;
    required uint32 last_login_time = 6;
    required uint32 last_notification_id = 7;
    required uint32 unprocessed_message_count = 8;
}
